qutip/core/operators.py

```diff
@@ -767,12 +765,13 @@ def qeye_like(qobj):
         Identity operator Qobj.
 
     """
-    from .cy.qobjevo import QobjEvo
-    if isinstance(qobj, QobjEvo):
-        qobj = qobj(0)
+    if qobj.shape[0] != qobj.shape[1]:
+        raise ValueError(
+            "Can't create an identity matrix like a non square matrix."
+        )
```

Comment:
Hasn't the error already been raised in the data layer? Such as in `identity_like_dense`

Reply:
It no longer use the the data layer `identity_like`.